hack/monitor_prow.py

```diff
@@ -42,28 +49,94 @@ def highlight(log_dir, dc):
     error = '"level":"error"'
     fatal = '"level":"fatal"'
     log = '{}/{}.log'.format(log_dir, dc)
-    header = '{}{}:{}'.format(BOLD, dc, RESET)
     while True:
+        debug("deployment/{}: gathering info".format(dc))
+        header = renderHeader(dc)
         lines = []
+        log_lines = []
         for pod in exec_cmd(
                 'oc', 'get', 'pods', '--namespace', 'ci',
                 '--selector', 'component={}'.format(dc),
                 '--output', 'jsonpath={.items[*].metadata.name}').split():
+            debug("deployment/{}: pod/{}: gathering info".format(dc, pod))
+            lines.extend(renderFlavor(pod, dc))
             cmd = [
                 'oc', 'logs', '--namespace', 'ci', '--since', '20m',
                 'pod/{}'.format(pod)]
             if pod.startswith('deck-internal'):
                 cmd += ['--container', 'deck']
+            debug("deployment/{}: pod/{}: getting logs".format(dc, pod))
             for l in exec_cmd(*cmd).splitlines():
                 if warn in l:
-                    lines.append(YELLOW + l + RESET)
+                    log_lines.append(YELLOW + l + RESET)
                 elif error in l or fatal in l:
-                    lines.append(RED + l + RESET)
+                    log_lines.append(RED + l + RESET)
+
+        if not log_lines and not lines:
+            header = header + " " + GREEN + "OK" + RESET
```

Comment:
Pythonista in me prefers `"".join()` or `format()` or string interpolation here. Most of the code here uses `format`, I guess.